### PR TITLE
merge conflicts resolved

### PR DESCRIPTION
**Pull request by asheremet**

### CODE REVIEW

File: translators/generator.js [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+function updateList(existingList, fn) {
+	var translators = [];
+	var sPath = path.join(repoPath, zoteroListPath);
+	fs.readdir(sPath, function(err, files) {
+		async.eachLimit(files, 10, function(file, cb) {
+			if(path.extname(file) == ".js"){
+				fs.readFile(path.join(sPath, file), function(err, data) {
+					var fileContent = data.toString();
+					if(fileContent.match(/(^\s*{[\s\S]*?\n})/)) {
+						var transObj = JSON.parse(RegExp.$1);
```
Comment (aepheus) [inline translators/generator.js]: might want some error catching here in case we get something that's not valid

File: translators/generator.js [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+						if((transObj.translatorType == 4 ) &&
```
Comment (aepheus) [inline translators/generator.js]: We may also need type 3 for all translators to work correctly. 3 is import translators (things like RIS or RW tagged)

File: translators/generator.js [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+							(transObj.browserSupport && transObj.browserSupport.indexOf('b') > -1)) {
+							translators.push({
+								translatorID: transObj.translatorID,
+								target: transObj.target,
+								priority: transObj.priority
+							});
+							putObjectToS3(translatorskey + transObj.translatorID + ".js", fileContent, function() {
```
Comment (aepheus) [inline translators/generator.js]: Need to give read permission to everyone.

File: translators/generator.js [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+	var sPath = path.join(repoPath, zoteroListPath);
+	fs.readdir(sPath, function(err, files) {
+		async.eachLimit(files, 10, function(file, cb) {
+			if(path.extname(file) == ".js"){
+				fs.readFile(path.join(sPath, file), function(err, data) {
+					var fileContent = data.toString();
+					if(fileContent.match(/(^\s*{[\s\S]*?\n})/)) {
+						var transObj = JSON.parse(RegExp.$1);
+						if((transObj.translatorType == 4 ) &&
+							(transObj.browserSupport && transObj.browserSupport.indexOf('b') > -1)) {
```
Comment (aepheus) [inline translators/generator.js]: this could blow up if browserSupport is undefined.

File: translators/generator.js [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+				})
+			}
+			else {
+				cb();
+			}
+		}, function() {
+			var unsupported = _.difference(existingList, _.map(translators, function(trans) {
+				return trans.translatorID.replace(".js", "")
+			}));
+			putObjectToS3(masterkey, "Zotero.TranslatorMasterList = " + JSON.stringify(translators), function() {
```
Comment (aepheus) [inline translators/generator.js:103]: should add read to everyone here as well.

File: translators/generator.js (hunk 4 above)
Comment (asheremet) [inline translators/generator.js]: why?

Comment (aepheus) [inline translators/generator.js]: indexOf on undefined. in the zotero code, I've seen checks that set a default for it, better safe than sorry.

Comment (asheremet) [inline translators/generator.js]: it checks for transObj.browserSupport first. Am I missing something?

Comment (aepheus) [inline translators/generator.js]: Oh, right. I don't know what I was thinking...

**Pull request merged**